protocol/contracts/beanstalk/sun/SeasonFacet/Distribution.sol

```diff
@@ -0,0 +1,172 @@
+            if (!capExceeded) break;
+        }
+
+        // Ship it.
+        for (uint256 i; i < shipmentAmounts.length; i++) {
+            if (shipmentAmounts[i] == 0) continue;
+            // Success is not verified here. Receive functions should be designed to never revert.
+            // address(this).call(
+            //     abi.encodeWithSelector(
+            //         bytes4(keccak256("receive(uint8,uint256,bytes)")),
+            //         shipmentRoutes[i].recipient,
+            //         shipmentAmounts[i],
+            //         shipmentRoutes[i].data
+            //     )
+            // );
+            receiveShipment(
+                shipmentRoutes[i].recipient,
+                shipmentAmounts[i],
+                shipmentRoutes[i].data
+            );
+        }
+
+        emit Ship(s.season.current, shipmentRoutes, shipmentAmounts);
+    }
+
+    /**
+     * @notice Determines the amount of Beans to distribute to each shipping route based on points.
+     * @dev Does not factor in route cap.
+     * @dev If points are 0, does not alter the associated shippingAmount.
+     * @dev Assumes shipmentAmounts and shipmentRoutes have matching shape and ordering.
+     */
+    function getBeansFromPoints(
+        uint256[] memory shipmentAmounts,
+        ShipmentPlan[] memory shipmentPlans,
+        uint256 totalPoints,
+        uint256 beansToShip
+    ) private pure {
+        for (uint256 i; i < shipmentPlans.length; i++) {
+            // Do not modify amount for streams with 0 points. They either are zero or have already been set.
+            if (shipmentPlans[i].points == 0) continue;
+            shipmentAmounts[i] = (beansToShip * shipmentPlans[i].points) / totalPoints; // round down
+        }
+    }
+
+    function hashShipmentRoute(
+        Storage.ShipmentRoute memory shipmentRoute
+    ) internal pure returns (bytes32 hash) {
+        return
+            keccak256(
+                abi.encode(
+                    shipmentRoute.planContract,
+                    shipmentRoute.planSelector,
+                    uint8(shipmentRoute.recipient),
+                    keccak256(shipmentRoute.data)
+                )
+            );
+    }
+
+    /**
+     * @notice Gets the shipping plan for all shipping routes.
+     * @dev Determines which routes are active and how many Beans they will receive.
+     * @dev GetPlan functions should never fail/revert. Else they will have no Beans allocated.
+     */
+    function getShipmentPlans(
+        Storage.ShipmentRoute[] memory shipmentRoutes
+    ) private view returns (ShipmentPlan[] memory shipmentPlans, uint256 totalPoints) {
+        shipmentPlans = new ShipmentPlan[](shipmentRoutes.length);
+        for (uint256 i; i < shipmentRoutes.length; i++) {
+            (bool success, bytes memory returnData) = shipmentRoutes[i].planContract.staticcall(
+                abi.encodeWithSelector(shipmentRoutes[i].planSelector, shipmentRoutes[i].data)
```

Comment:
good callout, but in this particular case it is incompatible bc we need to hold the selector in the plan object. `abi.encodeCall` requires a "function pointer" instead of bytes4. i do not see a way to convert bytes4 into function pointer.